spec/unit/resource/powershell_script_spec.rb

```diff
@@ -47,81 +47,11 @@
-      resource.guard_interpreter(:powershell_script)
-      allow_any_instance_of(Chef::GuardInterpreter::ResourceGuardInterpreter).to receive(:evaluate_action).and_return(false)
-      resource.only_if("echo hi")
-    end
-
-    it "enables convert_boolean_return by default for guards in the context of powershell_script when no guard params are specified" do
-      allow_any_instance_of(Chef::GuardInterpreter::ResourceGuardInterpreter).to receive(:evaluate_action).and_return(true)
-      allow_any_instance_of(Chef::GuardInterpreter::ResourceGuardInterpreter).to receive(:block_from_attributes).with(
-        { convert_boolean_return: true, code: "$true" }
-      ).and_return(Proc.new {})
-      resource.only_if("$true")
-    end
-
-    it "enables convert_boolean_return by default for guards in non-Chef::Resource::Script derived resources when no guard params are specified" do
-      node = Chef::Node.new
-      run_context = Chef::RunContext.new(node, nil, nil)
-      file_resource = Chef::Resource::File.new("idontexist", run_context)
-      file_resource.guard_interpreter :powershell_script
-
-      allow_any_instance_of(Chef::GuardInterpreter::ResourceGuardInterpreter).to receive(:block_from_attributes).with(
-        { convert_boolean_return: true, code: "$true" }
-      ).and_return(Proc.new {})
-      resource.only_if("$true")
-    end
-
-    it "enables convert_boolean_return by default for guards in the context of powershell_script when guard params are specified" do
-      guard_parameters = { cwd: "/etc/chef", architecture: :x86_64 }
-      allow_any_instance_of(Chef::GuardInterpreter::ResourceGuardInterpreter).to receive(:block_from_attributes).with(
-        { convert_boolean_return: true, code: "$true" }.merge(guard_parameters)
-      ).and_return(Proc.new {})
-      resource.only_if("$true", guard_parameters)
-    end
-
-    it "passes convert_boolean_return as true if it was specified as true in a guard parameter" do
-      guard_parameters = { cwd: "/etc/chef", convert_boolean_return: true, architecture: :x86_64 }
-      allow_any_instance_of(Chef::GuardInterpreter::ResourceGuardInterpreter).to receive(:block_from_attributes).with(
-        { convert_boolean_return: true, code: "$true" }.merge(guard_parameters)
-      ).and_return(Proc.new {})
-      resource.only_if("$true", guard_parameters)
-    end
-
-    it "passes convert_boolean_return as false if it was specified as true in a guard parameter" do
-      other_guard_parameters = { cwd: "/etc/chef", architecture: :x86_64 }
-      parameters_with_boolean_disabled = other_guard_parameters.merge({ convert_boolean_return: false, code: "$true" })
-      allow_any_instance_of(Chef::GuardInterpreter::ResourceGuardInterpreter).to receive(:block_from_attributes).with(
-        parameters_with_boolean_disabled
-      ).and_return(Proc.new {})
-      resource.only_if("$true", parameters_with_boolean_disabled)
-    end
+    expect(inherited_difference).to eq([])
   end
```

Comment:
I tried to figure out a way to make these tests meaningful for a while and it didn't seem possible without being very tied to the implementation of `Conditional`.

These tests currently do not set any expectations so there is no way they could fail. If they did set expectations on how `ResourceGuardInterpreter` worked, they would be failing now because calling `only_if` does not evaluate any `Conditional`s, it just gets them ready for later evaluation.